src/coffee/utils.coffee

```diff
@@ -1,4 +1,5 @@
 _ = require 'underscore'
+appRoot = require('app-root-path')
```

Comment:
Why adding a new dependency when you could just directly read the package.json?

E.G.: https://github.com/commercetools/nodejs/blob/59cb4a79f778efa4d4c4485c56311c1d0285ba65/packages/category-exporter/src/main.js

Reply:
this is traditionally a pain in the ass in NodeJS - there is no proper way to do it, see https://stackoverflow.com/questions/10265798/determine-project-root-from-a-running-node-js-application  - yes blame JavaScript ;)  Especially if you have to deal with cross-compilation like CoffeeScript - where the hell does my compiled JS end up? I have no clue at coding time ;) 

Adding this dependency is the lazy developer way to do it, plus this dependency doesn't introduce new ones, so I think it's ok to add it.

Reply:
I agree with Jens here. Looking at the spec file we can see how we import the built util file here:
https://github.com/sphereio/sphere-node-sdk/blob/c0113d74a7555644dd98e7ad7eaef284f0c847c1/src/spec/utils.spec.coffee#L2

which means we should be able to import it here as:
```
getVersion: () ->
  pjson = require '../package.json'
```

and inside the spec as
```
it "should return the current version of the sphere node sdk", ->
  pjson = require '../package.json'
```

Reply:
hmm I don't like this.. the coding time file system does not match the runtime file system..

Reply:
`util.coffee` is in `./src/coffee/...` - so at coding time I would have to assume that the `package.json` lives in `../../package.json` , after cross compilation the file ends up in `./lib` and therefore `../package.json` works, you see what I mean? So this code works incidentally, but it's not what I would call "systematic" because it assumes something about the file system which is configured in a build script. If I would change the build script, this library would break all over the place.

Reply:
I wouldn't call it incidental. It's how the mapping is set up in the Grunt/config file here:
https://github.com/sphereio/sphere-node-utils/blob/b3c15aea8fb8fd85e2356b87082d6c62bb53465e/Gruntfile.coffee#L28-L35

Meaning, you're right that we need to check how we've mapped the compiler in the project before starting to import things outside of the `src` folder. But that is a common procedure in node projects nowadays IMO. The same thing happens when using TypeScript/Flow/Babel.

It does, also, follow the pattern used in other files requiring modules from the outside.

E.g.
https://github.com/sphereio/sphere-node-utils/blob/4b691775646724af748d4cdd1e21d3467bec0253/src/spec/exports.spec.coffee#L1
https://github.com/sphereio/sphere-node-utils/blob/3b38c6393d55585df5b0a5d5b8a36f6be574b591/src/spec/helpers/project-credentials-config.spec.coffee#L3
https://github.com/sphereio/sphere-node-utils/blob/b3c15aea8fb8fd85e2356b87082d6c62bb53465e/src/spec/helpers/sftp.spec.coffee#L5-L6

Reply:
Well, in my humble opinion, it's a strong coupling between code and the way the lib is packaged, and therefore is detrimental to coupling. 
But fine I will do it, under protest ;)

Reply:
Ideally, I think it would be best if Grunt also supported some sort of module resolution. That way we simply wouldn't have to care about this at all and could always use relative paths from the coffee code.

Anyway, this is such old legacy repos that I bet no one would like to dive in to try to solve it 😄

Reply:
Be careful. We still use it in many production projects and it works ;)

Reply:
NPM has built in resolution of package.json properties by adding everything as a environment variable...   but again this is a runtime dependency...

Reply:
in the "good old times", we added stuff like version number during build time in properties files by the way

Reply:
ok it's done in the new version